src/Microsoft.ML.Core/Data/DataKind.cs

```diff
@@ -9,6 +9,30 @@ namespace Microsoft.ML.Data
     /// <summary>
     /// Specifies a simple data type.
     /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    /// Some transforms use the default value and/or missing value of the data types.
+    /// The table below shows the default value definition for each of the data types.
+    ///
+    /// | Type | Default Value | IsDefault Indicator |
+    /// | -- | -- | -- |
+    /// | <xref:Microsoft.ML.Data.DataKind.String> or [text](xref:Microsoft.ML.Data.TextDataViewType) | Empty or `null` string (both result in empty `System.ReadOnlyMemory<char>` | <xref:"System.ReadOnlyMemory`1.IsEmpty*> |
+    /// | [Key](xref:Microsoft.ML.Data.KeyDataViewType) type (supported by the unsigned integer types in `DataKind`) | Not defined | Always `false` |
+    /// | All other types | Default value of the corresponding system type as defined by .NET standard. In C#, default value expression `default(T)` provides that value. | Equality test with the default value |
+    ///
+    /// The table below shows the missing value definition for each of the data types.
+    ///
+    /// | Type | Missing Value | IsMissing Indicator |
+    /// | -- | -- | -- |
+    /// | <xref:Microsoft.ML.Data.DataKind.String> or [text](xref:Microsoft.ML.Data.TextDataViewType) | Not defined | Always `false` |
+    /// | [Key](xref:Microsoft.ML.Data.KeyDataViewType) type (supported by the unsigned integer types in `DataKind`) | `0` | Equality test with `0` |
```

Comment:
Could you add a in-memory example to explain `Key`'s missing value? `0` is probably a key value only exists in ML.NET and user can't really see it by using `CreateEnumerable`.

Reply:
I think this would be P2. I added it to backlog: [https://github.com/dotnet/machinelearning/issues/3832](https://github.com/dotnet/machinelearning/issues/3832)

---
In reply to: [290926731](https://github.com/dotnet/machinelearning/pull/3831#discussion_r290926731) [](ancestors = 290926731)